src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
```

Comment:
Refactor code:

The same suggestion as before, adding more meaningful names to the iterators would make this code more readable.
![image](https://github.com/pedro-ruas/mate-v2-js_average-ages/assets/54222238/b8387698-4e74-4dcb-8848-b44a53ee1866)